### PR TITLE
Process parses command

### PR DESCRIPTION
**Pull request by som-snytt**
Pasted parser and test into `sys.process`.

Did not cope with comment about backslash on Windows.

Fixes scala/bug#7947

### CODE REVIEW

**Comment by SethTisue** [comment]
`Runtime.exec` splits on spaces somehow, can't we call the same code?

**Comment by som-snytt** [comment]
No. It uses `StringTokenizer`, which is essentially deprecated as useless. The point here is to offer quoting, which is what the people (quite reasonably) want.

Another question is whether some such functionality should be promoted to `util`, but that is out of scope.

File: src/library/scala/sys/process/Parser.scala [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ * additional information regarding copyright ownership.
+ */
+
+package scala.sys.process
+
+import scala.annotation.tailrec
+
+/** A simple enough command line parser.
+ */
+private[scala] object Parser {
```
Comment (He-Pin) [inline src/library/scala/sys/process/Parser.scala:19]: grate.

**Review by lrytz — APPROVED** [review]

**Comment by som-snytt** [comment]
The scaladoc never mentioned that it splits the string.

**Comment by som-snytt** [comment]
Just updated jline3. The last commit:
```
Fix quote parsing and escaping, fixes #331
```

**Pull request merged**